### PR TITLE
LPS-165561 Use application default credentials if service account key is not provide

### PR DESCRIPTION
**Pull request by brandizzi**
# References

* This PR is a follow-up to https://github.com/liferay-core-infra/liferay-portal/pull/2005
* LPS Ticket: [LPS-165561](https://issues.liferay.com/browse/LPS-165561)
* LCD Ticket: [LCD-14380](https://issues.liferay.com/browse/LCD-14380)
* How it was tested: [Tests done](https://github.com/liferay-core-infra/liferay-portal/pull/2005#issuecomment-1278113769) (There are testing instructions in the LPS ticket, too.)

# Background

When no credentials are explicitly set, Google's libraries automatically attempt to federate access control as follows:

* If the environment variable `GOOGLE_APPLICATION_CREDENTIALS` is set, ADC uses the service account key or configuration file that the variable points to.
* If the environment variable `GOOGLE_APPLICATION_CREDENTIALS` isn't set, ADC uses the service account that is attached to the resource that is running your code. (workload identity)

As a best practice, Liferay Cloud is moving to leverage Workload Identity for access control across our services that interact with GCP directly. As we are now looking to move our Document Library to GCSStore for all customers, our Liferay service will now be doing just that.

This change can thus also be used by anyone running Liferay in a GKE and is not strictly limited to our use case in Cloud.

# Important note

This pull request has one more change: we also made `aes256Key` optional. This attribute was already optional, but we were forced to add it as an empty value to all files. We suspect it was not the intended behavior, so I removed it here. If there is some reason to make `aes256Key` mandatory, even if empty, please let us know!

### CODE REVIEW

File: modules/apps/portal-store/portal-store-gcs/src/test/java/com/liferay/portal/store/gcs/GCSStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	public void testGoogleCredentialsNull() throws Exception {
+		mockServiceAccountKey(null);
+
+		_gcsStore.activate(Collections.emptyMap());
+
+		assertGoogleCredentials(
+			ServiceAccountCredentials.getApplicationDefault());
+	}
+
+	protected void assertGoogleCredentials(
```
Comment (dantewang) [inline modules/apps/portal-store/portal-store-gcs/src/test/java/com/liferay/portal/store/gcs/GCSStoreTest.java:108]: @briangreenwald Please make these protected methods private.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/portal-store/portal-store-gcs/src/test/java/com/liferay/portal/store/gcs/GCSStoreTest.java (hunk 1 above)
Comment (dantewang) [reply modules/apps/portal-store/portal-store-gcs/src/test/java/com/liferay/portal/store/gcs/GCSStoreTest.java:108]: @brandizzi

Comment (brandizzi) [reply modules/apps/portal-store/portal-store-gcs/src/test/java/com/liferay/portal/store/gcs/GCSStoreTest.java:108]: @dantewang Done! Changes in https://github.com/liferay-core-infra/liferay-portal/pull/2170.

**Comment by brandizzi** [comment]
Re-sent as https://github.com/liferay-core-infra/liferay-portal/pull/2170.

**Pull request closed without merging**